lib/carrierwave/base64/adapter.rb

```diff
@@ -34,10 +34,53 @@ def mount_base64_uploader(attribute, uploader_class, options = {})
 
           super Carrierwave::Base64::Base64StringIO.new(data.strip, filename)
         end
-        # rubocop:enable Metrics/MethodLength, Metrics/AbcSize
-        # rubocop:enable Metrics/CyclomaticComplexity
-        # rubocop:enable Metrics/PerceivedComplexity
       end
+
+      def mount_base64_uploaders(attribute, uploader_class, options = {})
+        mount_uploaders attribute, uploader_class, options
+
+        if options[:file_name].is_a?(String)
+          warn(
+            '[Deprecation warning] Setting `file_name` option to a string is '\
+            'deprecated and will be removed in 3.0.0. If you want to keep the '\
+            'existing behaviour, wrap the string in a Proc'
+          )
+        end
+
+        define_method "#{attribute}=" do |data_array|
+          return if data_array == send(attribute).to_s
+
+          if respond_to?("#{attribute}_will_change!") && data_array.present?
+            send "#{attribute}_will_change!"
+          end
+
+          return super(data_array) unless data_array.is_a?(Array)
+
+          results = data_array.map do |data|
```

Comment:
why `map` if you're not using the resulting array? You can just use `each` here, and have `key` and `value` as arguments, since it's a hash actually